### ISSUE TITLE
volume support in v2.0 Preview

### ISSUE THREAD

**Issue by ianmjones** [rancher/rancher#9949]
**Rancher versions:**
rancher/server:preview
rancher/agent:v2.0-alpha4

**Infrastructure Stack versions:**
Holy cow, too many to list!

**Docker version: (`docker version`,`docker info` preferred)**
Containers: 23
 Running: 21
 Paused: 0
 Stopped: 2
Images: 14
Server Version: 17.03.2-ce
Storage Driver: overlay
 Backing Filesystem: extfs
 Supports d_type: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins: 
 Volume: local longhorn
 Network: bridge host macvlan null overlay
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: 4ab9917febca54791c5f071a9d1f404867857fcc
runc version: 54296cf40ad8143b62dbcaa1d90e520a2136ddfe
init version: 949e6fa
Security Options:
 seccomp
  Profile: default
Kernel Version: 4.9.45-rancher
Operating System: RancherOS v1.1.0
OSType: linux
Architecture: x86_64
CPUs: 2
Total Memory: 3.59 GiB
Name: ians-mac-mini
ID: IAHA:HERM:27BR:D2I6:AV7K:YYGH:ZNGP:THVA:EPGI:5FHH:KKO3:THO2
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
Experimental: false
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false

**Operating system and kernel: (`cat /etc/os-release`, `uname -r` preferred)**
NAME="RancherOS"
VERSION=v1.1.0
ID=rancheros
ID_LIKE=
VERSION_ID=v1.1.0
PRETTY_NAME="RancherOS v1.1.0"
HOME_URL="http://rancher.com/rancher-os/"
SUPPORT_URL="https://forums.rancher.com/c/rancher-os"
BUG_REPORT_URL="https://github.com/rancher/os/issues"
BUILD_ID=

**Type/provider of hosts: (VirtualBox/Bare-metal/AWS/GCE/DO)**
Bare-metal

**Setup details: (single node rancher vs. HA rancher, internal DB vs. external DB)**
Internal DB (as per http://rancher.com/docs/rancher/v2.0/en/quick-start-guide/)

**Environment Template: (Cattle/Kubernetes/Swarm/Mesos)**
Kubernetes!

**Steps to Reproduce:**
1. Add 3 hosts.
1. Add custom catalog "Ian's Catalog" from https://github.com/ianmjones/rancher-catalog.git
1. Apps -> Launch From Catalog
1. Pick "ActorDB"
1. Save (defaults are fine for 3 hosts)

**Results:**
Main "actordb" service does not start, many errors logged to service log similar to following:
```
SQL [insert into `volume` (`name`, `account_id`, `kind`, `uuid`, `state`, `created`, `data`, `deployment_unit_id`, `environment_id`, `volume_template_id`) values (?, ?, ?, ?, ?, ?, ?, ?, ?, ?)]; (conn:227) Field 'cluster_id' doesn't have a default value Query is: insert into `volume` (`name`, `account_id`, `kind`, `uuid`, `state`, `created`, `data`, `deployment_unit_id`, `environment_id`, `volume_template_id`) values (?, ?, ?, ?, ?, ?, ?, ?, ?, ?), parameters ['actordb_actordb-data_2_fc0e0',5,'volume','1a5d669d-b561-4eb7-a501-ffac63756a74','requested','2017-09-26 20:45:49.468','{"fields":{"driver":"local","driverOpts":{}}}',30,37,1]
```

I'm guessing this is related to the ActorDB stack having been built for Cattle, for whatever reason it uses something that does not translate correctly to new Kubernetes based v2.0 setup.

By default it just uses "local" volume driver, "per_container" scope.

Stack is publicly available at https://github.com/ianmjones/rancher-catalog/tree/master/templates/actordb/0, it's not very big, one service with global LB.

**Comment by deniseschannon** [comment]
Only host bind mounted volumes are working in v2.0.0-alpha6 as noted in the release notes. This is on the roadmap to be fixed.

**Comment by deniseschannon** [comment]
Also, please make sure to use the `preview` tag, which is currently pointed at `v2.0.0-alpha6`. `alpha4` was not stable enough to be used for `preview`. :)

**Comment by ianmjones** [comment]
Thanks @deniseschannon.

> Only host bind mounted volumes are working in v2.0.0-alpha6 as noted in the release notes. This is on the roadmap to be fixed.

Ah, missed that, thanks for letting me know.

> Also, please make sure to use the preview tag, which is currently pointed at v2.0.0-alpha6. alpha4 was not stable enough to be used for preview. :)

Hmmm, I'm using `preview` for the server, but `rancher/agent:v2.0-alpha4` came from the given command to copy when adding a custom host. Would be good if that defaulted to `preview` too.

**Comment by deniseschannon** [comment]
Oops! I just saw the alpha4 and thought that was for the Rancher server. That is the correct version of rancher/agent to be using!

**Comment by vincent99** [comment]
TL;DR: we know storage doesn't work yet (and catalogs), I don't think we need another issue to track that.
